PERFORMANCE.md

```diff
@@ -0,0 +1,343 @@
+
+#### Unpack Tensor (retrieve the inference result)
+![Inference Colo](/figures/unpack_tensor_inf_std.png "Inference Colocated")
+
+## Colocated Inference
+
+The following are colocated deployment scaling results from the cpp-inference and fortran-inference scaling tests with ResNet-50 and the imagenet dataset. For more information on these scaling tests, please see
+the SmartSim paper on [arXiv](https://arxiv.org/pdf/2104.09355.pdf).
+
+#### Inference Colocated Run Configuration File
+```bash
+[run]
+name = run-2023-08-17-18:23:38
+path = results/inference-colocated-scaling/run-2023-08-17-18:23:38
+smartsim_version = 0.5.0
+smartredis_version = 0.3.1
+db = redis-server
+date = 2023-08-17
+language = ['cpp', 'fortran']
+
+[attributes]
+colocated = 1
+client_per_node = [18]
+client_nodes = [4, 8, 12, 16]
+database_cpus = [8]
+database_port = 6780
+batch_size = [96]
+device = GPU
+num_devices = 1
+iterations = 100
+language = ['cpp', 'fortran']
+node_feature = {'constraint': 'P100'}
+```
+
+#### Put Tensor (send image to database)
+![Inference Colo](/figures/put_tensor_inf_colo.png "Inference Colocated")
+
+#### Run Script (preprocess image)
+![Inference Colo](/figures/run_script_inf_colo.png "Inference Colocated")
+
+#### Run Model (run resnet50 on the image)
+![Inference Colo](/figures/run_model_inf_colo.png "Inference Colocated")
+
+#### Unpack Tensor (retrieve the inference result)
+![Inference Colo](/figures/unpack_tensor_inf_colo.png "Inference Colocated")
+
+## Inference Performance Analysis
+
+In this section, we will compare inference clients: `put-tensor`, `unpack-tensor`, `run_model` and `run_script`
+for colocated and standard deployment.
+
+> Note that Inference is the process of running data points into a machine learning model to calculate an output such as a single numerical score. The SmartSim-Scaling tests use Pytorch's implementation of Resnet50 to feed 
+the ImageNet dataset through. We put the ImageNet data into the database via put_tensor, process the data via run_script, run a machine learning model with the data via run_model then retrieve the data via unpack_tensor.
+
+- `put-tensor` : Colocated deployment offers a consistent median for put_tensor times. Standard deployment shows a slight
+increase in median as client count grows. However, due to machine constraints, colocated is maxed at 288 clients while
+standard maxes at 1800 clients. We can conclude that there is not a significant performance hit putting information into the database when comparing standard and colocated.
+
+- `run_script` : Colocated deployment offers a faster run_script client than standard deployment. We can 
+infer colocated deployment is able to transfer information faster when processing data than standard deployment. 
+This is likely because communication time is cut when using colocated deployment. There are also not as many requests sent using colocated deployment than standard which result in faster completion for colocated.
```

Comment:
Why do you think the number of requests is different between colo and standard? The real different is that we get rid of network latency with colo